internal/commands/generate.go

```diff
@@ -5,50 +5,62 @@ import (
 	"io/ioutil"
 	"os"
 	"path"
-	"strings"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/plexsystems/promdoc/internal/rendering"
 )
 
 // NewGenerateCommand creates a new generate command
 func NewGenerateCommand() *cobra.Command {
 	cmd := cobra.Command{
-		Use:   "generate <output-dir>",
+		Use:   "generate",
 		Short: "Generate documentation from a given folder",
-		Args:  cobra.ExactArgs(1),
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := viper.BindPFlag("out", cmd.Flags().Lookup("out")); err != nil {
+				return fmt.Errorf("bind out flag: %w", err)
+			}
+			return nil
+		},
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := runGenerateCommand(args[0]); err != nil {
+			if err := runGenerateCommand(); err != nil {
 				return fmt.Errorf("generate: %w", err)
 			}
 
 			return nil
 		},
 	}
 
+	cmd.Flags().StringP("out", "o", "alerts.md",
+		"file name or path for the output-file")
+
 	return &cmd
 }
 
-func runGenerateCommand(outputFile string) error {
+func runGenerateCommand() error {
+	out := viper.GetString("out")
+	outputDir, outputFile := path.Split(out)
 	workingDir, err := os.Getwd()
 	if err != nil {
 		return fmt.Errorf("get working dir: %w", err)
 	}
-
-	fileTokens := strings.Split(outputFile, ".")
-	if len(fileTokens) == 0 {
-		return fmt.Errorf("get file extension: %w", err)
+	if outputDir == "" {
```

Comment:
While I do like trying to leverage the standard library, in this case the `Split` func--the current behavior isn't _quite_ what we're looking for.

```shell
promdoc generate --out docs
```

Should create an `alerts.md` into the `docs` directory. Unfortunately, Split will consider "docs" to be a file.

Reply:
should we consider to treat all paths that do not have an extension to be directories?

Reply:
Yep! If it's a file, it should have an extension that we can use to figure out what type of file to render. Otherwise it should be assumed to be a directory and output alerts.md (markdown).